### PR TITLE
GO AQL queries

### PR DESCRIPTION
**Pull request by kellyhuang21**
added AQL queries for GO as yaml files

### CODE REVIEW

File: GO_stored_queries/GO_get_hierarchicalAncestors.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  type: object
+  required: [key]
+  properties:
+    key:
+      type: string
+      title: Document key
+      description: Get all hierarchical ancestors for this term
+query: |
+  WITH GO_test_term FOR v, e IN 1..1000000 OUTBOUND @key GO_test_edges_isa, GO_test_edges_relationship
+      RETURN {term: v, edge: e}
```
Comment (jayrbolton) [inline GO_stored_queries/GO_get_hierarchicalAncestors.yaml]: Should we filter on relationship type? I'm reading this comment: `Hierarchical ancestors include is-a and other related parents, such as part-of/develops-from, that imply a hierarchical relationship` -- and I'm thinking not all relationships are hierarchical, so we need to filter those out.

We might need two separate queries that we return on a union: isa query, and a separate relationship query that filters based on `.relationship_type`.

File: GO_stored_queries/GO_get_ancestors.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# Get all ancestors (all parent's parents) for this term
+
+name: GO_get_ancestors
+params:
+  type: object
+  required: [key]
+  properties:
```
Comment (jayrbolton) [inline GO_stored_queries/GO_get_ancestors.yaml]: Let's add params here for `limit` and `offset` which can default to to 20 and 0 respectively. This way we 
In the query we'd add a line for `LIMIT @offset, @limit`
Let's also add a SORT BY clause in the query on the go term ID

File: GO_stored_queries/GO_get_children.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# Get the direct hierarchical children 
+# for this term. Hierarchical children include is-a 
+# and other related children, such as part-of/develops-from, 
+# that imply a hierarchical relationship 
+name: GO_get_children
+params:
+  type: object
+  required: [key]
+  properties:
+    key:
```
Comment (jayrbolton) [inline GO_stored_queries/GO_get_children.yaml]: Same comment as above: let's add limit, offset, and a sort by on go id

File: GO_stored_queries/GO_get_descendents.yaml [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+name: GO_get_descendants
+params:
+  type: object
+  required: [key]
+  properties:
+    key:
+      type: string
+      title: Document key
+      description: Get all descendents of this term
+query: |
```
Comment (jayrbolton) [inline GO_stored_queries/GO_get_descendents.yaml]: Same comment as above: let's add limit, offset, and a sort by on go id

File: GO_stored_queries/GO_get_ancestors.yaml (hunk 2 above)
Comment (jayrbolton) [reply GO_stored_queries/GO_get_ancestors.yaml]: Also, in the json schema for the "offset" property, add a "maximum" of 10000

File: GO_stored_queries/GO_get_hierarchicalChildren.yaml [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+name: GO_get_hierarchicalChildren
+params:
+  type: object
+  required: [key]
+  properties:
+    key:
+      type: string
+      title: Document key
+      description: Get the direct hierarchical children for this term
+query: |
```
Comment (jayrbolton) [inline GO_stored_queries/GO_get_hierarchicalChildren.yaml]: Same comment as above: let's add limit, offset, and a sort by on go id

File: GO_stored_queries/GO_get_hierarchicalDescendents.yaml [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+name: GO_get_hierarchicalDescendants
+params:
+  type: object
+  required: [key]
+  properties:
+    key:
+      type: string
+      title: Document key
+      description: Get all hierarhical descendents of this term
+query: |
```
Comment (jayrbolton) [inline GO_stored_queries/GO_get_hierarchicalDescendents.yaml]: Same comment as above: let's add limit, offset, and a sort by on go id

File: GO_stored_queries/GO_get_hierarchicalParents.yaml [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+name: GO_get_hierarchicalParents
+params:
+  type: object
+  required: [key]
+  properties:
+    key:
+      type: string
+      title: Document key
+      description: Get the direct hierarchical parent for this term 
+query: |
```
Comment (jayrbolton) [inline GO_stored_queries/GO_get_hierarchicalParents.yaml]: Same comment as above: let's add limit, offset, and a sort by on go id

File: GO_stored_queries/GO_get_parents.yaml [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+name: GO_get_parents
+params:
+  type: object
+  required: [key]
+  properties:
+    key:
+      type: string
+      title: Document key
+      description: Get the direct parents for a specific term 
+query: |
```
Comment (jayrbolton) [inline GO_stored_queries/GO_get_parents.yaml]: Same comment as above: let's add limit, offset, and a sort by on go id

File: GO_stored_queries/GO_get_related.yaml [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+params:
+  type: object
+  required: [key]
+  properties:
+    key:
+      type: string
+      title: Document key
+      description: Get all immediate related terms for this term
+query: |
+  WITH GO_test_term FOR v, e IN 1 ANY "GO_test_term/GO:0000022" GO_test_edges_isa
```
Comment (jayrbolton) [inline GO_stored_queries/GO_get_related.yaml]: Should this be relationship edges rather than isa?

File: GO_stored_queries/GO_get_related.yaml [hunk 10]
```diff
@@ -0,0 +1,13 @@
...
+name: GO_get_related
+params:
+  type: object
+  required: [key]
+  properties:
+    key:
+      type: string
+      title: Document key
+      description: Get all immediate related terms for this term
+query: |
```
Comment (jayrbolton) [inline GO_stored_queries/GO_get_related.yaml]: Same comment as above: let's add limit, offset, and a sort by on go id

File: GO_stored_queries/GO_get_siblings.yaml [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+name: GO_get_siblings
+params:
+  type: object
+  required: [key]
+  properties:
+    key:
+      type: string
+      title: Document key
+      description: Get all siblings of this term
+query: |
```
Comment (jayrbolton) [inline GO_stored_queries/GO_get_siblings.yaml]: Same comment as above: let's add limit, offset, and a sort by on go id

File: GO_stored_queries/GO_get_hierarchicalAncestors.yaml (hunk 1 above)
Comment (kellyhuang21) [reply GO_stored_queries/GO_get_hierarchicalAncestors.yaml]: Oh right. The hierarchical relationships seem to be has_part and part_of. Is occurs_in hierarchical?

Relationships in GO: ends_during, happens_during, has_part, negatively_regulates, never_in_taxon, occurs_in, part_of, positively_regulates, regulates, starts_during.

**Pull request merged**